src/acom_music_box/data_output.py

```diff
@@ -0,0 +1,65 @@
+import os
+import pandas as pd
+import xarray as xr
+
+class DataOutput:
+    def __init__(self, df, args):
+        self.df = df
+        self.args = args
+        self.unit_mapping = {
+            'ENV.temperature': 'K',
+            'ENV.pressure': 'Pa',
+            'ENV.number_density_air': 'kg -m3',
+            'time': 's'
+        }
+
+    def ensure_output_path(self):
+        """Ensure the output path is valid and create directories if needed."""
+        if os.path.dirname(self.args.output) == '':
+            self.args.output = os.path.join(os.getcwd(), self.args.output)
+        elif not os.path.basename(self.args.output):
+            raise ValueError(f"Invalid output path: '{self.args.output}' does not contain a filename.")
+        
+        dir_path = os.path.dirname(self.args.output)
+        if dir_path and not os.path.exists(dir_path):
+            os.makedirs(dir_path, exist_ok=True)
+
+    def append_units_to_columns(self):
```

Comment:
```suggestion
    def _append_units_to_columns(self):
```

Since this is a private function, we should prefix it with an underscores.